crescent/internal/handle_resp.py

```diff
@@ -196,6 +202,27 @@ def _context_from_interaction_resp(interaction: CommandInteraction) -> Context:
     )
 
 
+def _base_context_to_subclass(ctx: BaseContext, ctx_type: type[ContextT]) -> ContextT:
+    return ctx_type(
+        interaction=ctx.interaction,
+        app=ctx.app,
+        application_id=ctx.application_id,
+        type=ctx.type,
+        token=ctx.token,
+        id=ctx.id,
+        version=ctx.version,
+        channel_id=ctx.channel_id,
+        guild_id=ctx.guild_id,
+        user=ctx.user,
+        member=ctx.member,
+        command=ctx.command,
+        group=ctx.group,
+        sub_group=ctx.sub_group,
+        command_type=ctx.command_type,
+        options=ctx.options,
+    )
+
+
```

Comment:
Feels like this functionality belongs on either the BaseContext or the Context/AutocompleteContext?

Reply:
Hmmm, I could do a rusty `into()` method.